keystrokes/src/components/Character/CharacterDetails/ManualKeybindModal.js

```diff
@@ -0,0 +1,251 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Button, Grid, TextField, MenuItem } from '@material-ui/core';
+import update from 'immutability-helper';
+import PropTypes from 'prop-types';
+import KeybindTable from './KeybindTable';
+import styleGuide from '../../../stylesheets/style';
+import { verifyKey, validatePress } from '../../utils/utils';
+import {
+	ref,
+	targetting,
+	mods,
+	characterDetails,
+} from '../../../config/constants';
+
+ManualKeybindModal.propTypes = {
+	isOpen: PropTypes.bool.isRequired,
+	setIsOpen: PropTypes.func.isRequired,
+	characterClass: PropTypes.string.isRequired,
+	characterSpec: PropTypes.number.isRequired,
+	setAllKeybindings: PropTypes.func.isRequired,
+	allKeybindings: PropTypes.object.isRequired,
+	keyBindingKey: PropTypes.string.isRequired,
+};
+
+function ManualKeybindModal({
+	isOpen,
+	setIsOpen,
+	characterClass,
+	characterSpec,
+	setAllKeybindings,
+	allKeybindings,
+	keyBindingKey,
+}) {
+	const classes = styleGuide();
+	const [keybinding, setKeybinding] = useState({
+		Spell: null,
```

Comment:
curious why you went with UpperCase for variables here

Reply:
I think i was following a database structure, i think before when using SQL i would call things with a capital and just kept that convention.